var/spack/repos/builtin/packages/mysql/package.py

```diff
@@ -105,8 +119,34 @@ def cmake_args(self):
             options.append('-DWITHOUT_SERVER:BOOL=ON')
         return options
 
+    def fix_dtrace_usr_bin_path(self, spack_env):
+        # dtrace may cause build to fail because it uses
+        # '/usr/bin/python' in the shebang. To work around that we copy
+        # the original script into a temporary folder, and change the
+        # shebang to '/usr/bin/env python'.
+        dtrace = which('dtrace').path
+        dtrace_copy_path = os.path.join(tempfile.mkdtemp(), 'dtrace-copy')
```

Comment:
Please don't use `tempfile`.  Instead, create a file/folder in a predictable place in the Spack stage (i.e. in or close to where the package is already being built).  No need to create extra files elsewhere in the system; which I have a feeling could come back to bite us somewhere, somehow.

Reply:
Upon investigation, and subsequent discussion with @scheibelp on a telcon, I'd prefer to keep this as the temporary file for the following reasons:
* Spack assumes things about the structure of the stage area, and putting files at the top level can cause undesirable behavior (I believe a recent PR may have resolved this but I have not verified)
* I do not wish to pollute the expanded source area of the stage tree if at all possible as this workaround is not, strictly speaking, part of the build of the source as a patch might be, but rather a modification of the external environment as seen by the build.

The only negative consequences I can imagine from using `tempfile.mkdtemp()` would be if a user explicitly set `TMPDIR` to  an area mounted `noexec`. I have never seen any system where the standard `TMPDIR` location (`/var/tmp`) has been so marked. In the case of the user doing this, I think it would fit into the category of, "It's going to hurt when you do that."